### PR TITLE
Add support for Razer Viper 8KHz

### LINKED ISSUE #1426: Support for Razer Viper 8KHz
### Device Info

* Name: Viper 8KHz
* Product Number: RZ01-03580100
* Model Number: RZ01-0358

### Device Information
```
lsusb -d 1532: -v


Bus 001 Device 004: ID 1532:0091 Razer USA, Ltd Razer Viper 8KHz
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x0091 
  bcdDevice            1.01
  iManufacturer           1 Razer
  iProduct                2 Razer Viper 8KHz
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xe0
      Self Powered
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.00
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     159
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0010  1x 16 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
Device Qualifier (for other device speed):
  bLength                10
  bDescriptorType         6
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  bNumConfigurations      0
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0001
  Self Powered
```

```
ls -lah /dev/input/by-id/ | grep Raze
lrwxrwxrwx 1 root root  10 02-03 19:28 usb-Razer_Razer_Viper_8KHz-event-if01 -> ../event10
lrwxrwxrwx 1 root root  10 02-03 19:31 usb-Razer_Razer_Viper_8KHz-event-mouse -> ../event11
lrwxrwxrwx 1 root root   9 02-03 19:28 usb-Razer_Razer_Viper_8KHz-if01-event-kbd -> ../event5
lrwxrwxrwx 1 root root  10 02-03 19:11 usb-Razer_Razer_Viper_8KHz-if02-event-kbd -> ../event12
lrwxrwxrwx 1 root root   9 02-03 19:31 usb-Razer_Razer_Viper_8KHz-mouse -> ../mouse0
```

### Packet Captures

This is a 8000Hz mouse. I don't know how Windows (via Synapse) is switching it to 8000Hz. I guess USB2.0 is used (USB1 polling seems to be limited to 1kHZ), but HWINFO.exe did show only USB1 enabled when using 8000Hz.

I saved some profiles on the mouse (with 8kHz polling), but these seem not to work in the absence of Windows software, even if I assigned 'Cycle Up Profile' to one of the buttons, the profiles are not switching (I'd see LED changing color in such case).

I'll be willing to buy this mouse to someone else, if that person promises to do some good effort on trying to reverse engineer the USB1->USB2 switching protocol and implement it here (or via some script, it doesn't need to be in openrazer).

### PR DESCRIPTION
**Pull request by openglfreak**
Fixes #1426

### CODE REVIEW

**Comment by openglfreak** [comment]
^ no code changes, I only changed the commit messages

**Comment by z3ntu** [comment]
The poll rate *is* used by the daemon so support would very good but not strictly necessary for inclusion.

**Comment by openglfreak** [comment]
I'll implement it if someone figures out which packets to send. I didn't figure that out with trial & error.

**Comment by openglfreak** [comment]
Got it implemented, thanks to [this commit](https://github.com/sfroment/librazermacos/commit/74aac072ac7ee800cf53b439c459ae034a5b8c37#diff-f4e30a35643af9229934705075902f1b711153ce01d3c443b87b5d66314e412dR1018).

**Comment by openglfreak** [comment]
Why is that commit marked as unverified it's literally signed with the same key as the one before it

**Comment by openglfreak** [comment]
Looks like a bug, `git commit --amend --gpg-sign --reset-author` seems to only work correctly with `--no-edit`

File: daemon/openrazer_daemon/dbus_services/dbus_methods/mamba.py [hunk 1]
```diff
@@ -322,3 +322,41 @@ def get_poll_rate(self):
     self.logger.debug("DBus call get_poll_rate")
 
     return int(self.poll_rate)
+
+
+@endpoint('razer.device.misc', 'setPollRate', in_sig='q')
+def set_poll_rate2(self, rate):
```
Comment (z3ntu) [inline daemon/openrazer_daemon/dbus_services/dbus_methods/mamba.py]: Please use a more descriptive name like `set_poll_rate_8k`

Comment (z3ntu) [reply daemon/openrazer_daemon/dbus_services/dbus_methods/mamba.py]: Or even better, you can modify the original function to use `self.POLL_RATES` if available for checking.

**Review by z3ntu — CHANGES_REQUESTED** [review]
Looks pretty good, just one minor request right now.

File: daemon/openrazer_daemon/dbus_services/dbus_methods/mamba.py (hunk 1 above)
Comment (openglfreak) [reply daemon/openrazer_daemon/dbus_services/dbus_methods/mamba.py]: Yeah that's a much better solution, I implemented it in the new commit.

**Review by z3ntu — APPROVED** [review]
LGTM, thanks for the pull request!

**Comment by gaeldelameme** [comment]
Hey! How can i install the code of Viper 8Khz?

**Comment by gaeldelameme** [comment]
Hey, so I have to overwrite the source code to get the support for the viper 8Khz?

**Comment by lah7** [comment]
@gaeldelameme It depends on your distro. On Arch, install `openrazer-meta-git`. On Ubuntu, add the `openrazer/daily` PPA and update. For all others, grab a copy of this repository and [build a package](https://github.com/openrazer/openrazer/wiki/Building-a-package).

**Comment by AlexanderCurl** [comment]
> On Arch, install openrazer-meta-git.

I did, clean installed it, updated it, but no luck. I've used openrazer-meta-dkms-git also, still don't show my mouse.

**Comment by AlexanderCurl** [comment]
Okay I tested with evhz on Arch linux, it's reporting only 1000Hz maximum, hence it's set to 8000Hz in polychromatic.

`Razer Razer Viper 8KHz: Latest   972Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest   975Hz, Average   992Hz
Razer Razer Viper 8KHz: Latest  1024Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest  1002Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest   999Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest   982Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest  1019Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest  1000Hz, Average   993Hz
Razer Razer Viper 8KHz: Latest   500Hz, Average   985Hz
Razer Razer Viper 8KHz: Latest   977Hz, Average   985Hz
Razer Razer Viper 8KHz: Latest  1022Hz, Average   985Hz
Razer Razer Viper 8KHz: Latest   495Hz, Average   977Hz
Razer Razer Viper 8KHz: Latest  1012Hz, Average   977Hz
Razer Razer Viper 8KHz: Latest   334Hz, Average   966Hz
Razer Razer Viper 8KHz: Latest   905Hz, Average   965Hz
Razer Razer Viper 8KHz: Latest   342Hz, Average   955Hz
Razer Razer Viper 8KHz: Latest   143Hz, Average   942Hz

Average for Razer Razer Viper 8KHz:   942Hz
`

**Pull request merged**